src/main/java/net/mcreator/workspace/misc/WorkspaceInfo.java

```diff
@@ -117,11 +116,14 @@ public List<ModElement> getElementsOfType(ModElementType<?> type) {
 		}
 	}
 
-	public List<ModElement> getRecipesOfType(String typestring) {
+	public List<GeneratableElement> getModElements(String typestring) {
+		return getModElements(ModElementTypeLoader.getModElementType(typestring));
+	}
+
+	public List<GeneratableElement> getModElements(ModElementType<?> type) {
 		try {
-			return workspace.getModElements().parallelStream().filter(e -> e.getType() == ModElementType.RECIPE)
-					.filter(e -> e.getGeneratableElement() instanceof Recipe re && re.recipeType.equals(typestring))
-					.toList();
+			return workspace.getModElements().parallelStream().filter(e -> e.getType() == type)
+					.map(ModElement::getGeneratableElement).filter(Objects::nonNull).toList();

```

Comment:
This will call getGeneratableElement in parallel threads which has chance of running converters in parallel from different threads which is calling for trouble

Reply:
No running them in parallel or no calling `ModElement::getGeneratableElement`?

Reply:
Not running them in parallel, GEs are needed to obtain data from them

Reply:
:heavy_check_mark: Fixed

Reply:
I also noticed some other methods in this class use parallel streams too.

Reply:
The only one problematic is recipe check and is not used and I will remove it when I get time today

Others are thread-safe as MEs are thread-safe so I don't really see the problem here

If there is one, I stand corrected, of course ;)

Reply:
But not `getModElements()`?

Reply:
> MEs are thread-safe

Reply:
getModElements just returns list, and does not perform thread-unsafe actions such as getGeneratableElement